python/beeai_framework/workflows/workflow.py

```diff
@@ -61,12 +78,23 @@ class Workflow(Generic[T, K]):
 
     _RESERVED_STEP_NAMES: ClassVar = [START, SELF, PREV, NEXT, END]
 
+    emitter: Emitter
+
     def __init__(self, schema: type[T], name: str = "Workflow") -> None:
         self._name = name
         self._schema = schema
         self._steps: dict[K, WorkflowStepDefinition[T, K]] = {}
         self._start_step: K | None = None
 
+        # replace any non-alphanumeric char with _
+        formatted_name = re.sub(r"\W+", "_", self._name).lower()
+        self.emitter = Emitter.root().child(
+            EmitterInput(
```

Comment:
FYI this will need updating after https://github.com/i-am-bee/beeai-framework/pull/397 so which ever PR is merged second will need to address them

Reply:
https://github.com/i-am-bee/beeai-framework/pull/416/commits/59e0cfda9ccbc739ead81a749632b5bd1b3b3d68